terraform/modules/google-cloud/apps/client-monitor/templates/cloud-init.yaml

```
@@ -0,0 +1,208 @@
+#cloud-config
+
+write_files:
+  - path: /etc/dev.firezone.client/token
+    content: ${firezone_token}
+    permissions: "0600"
+    owner: root
+
+  - path: /etc/systemd/system/firezone.service
+    owner: root
+    content: |
+      [Unit]
+      Description=Firezone Client
+
+      [Service]
+      AmbientCapabilities=CAP_NET_ADMIN
+      CapabilityBoundingSet=CAP_CHOWN CAP_NET_ADMIN
+      DeviceAllow=/dev/net/tun
+      LockPersonality=true
+      MemoryDenyWriteExecute=true
+      NoNewPrivileges=true
+      PrivateMounts=true
+      PrivateTmp=true
+      PrivateUsers=false
+      ProcSubset=pid
+      ProtectClock=true
+      ProtectControlGroups=true
+      ProtectHome=true
+      ProtectHostname=true
+      ProtectKernelLogs=true
+      ProtectKernelModules=true
+      ProtectKernelTunables=true
+      ProtectProc=invisible
+      ProtectSystem=full
+      RestrictAddressFamilies=AF_INET AF_INET6 AF_NETLINK AF_UNIX
+      RestrictNamespaces=true
+      RestrictRealtime=true
+      RestrictSUIDSGID=true
+      StateDirectory=dev.firezone.client
+      SystemCallArchitectures=native
+      SystemCallFilter=@aio @basic-io @file-system @io-event @ipc @network-io @signal @system-service
+      UMask=077
+
+      Environment="FIREZONE_API_URL=${firezone_api_url}"
+      Environment="FIREZONE_DNS_CONTROL=systemd-resolved"
+      Environment="FIREZONE_ID=${firezone_client_id}"
+      Environment="RUST_LOG=${firezone_client_log_level}"
+      Environment="LOG_DIR=/var/log/firezone"
+
+      ExecStart=/usr/local/bin/firezone-linux-client standalone
+      Type=notify
+      User=root
+
+      [Install]
+      WantedBy=default.target
+
+  - path: /etc/google-cloud-ops-agent/config.yaml
+    permissions: "0644"
+    owner: root
+    content: |
+      logging:
+        receivers:
+          firezone_monitor:
+            type: files
+            include_paths:
+            - /var/log/firezone_monitor/*.log
+        service:
+          pipelines:
+            firezone_monitor_pipeline:
+              receivers: [firezone_monitor]
+
+  - path: /etc/cron.d/firezone_monitor
+    owner: root
+    content: |
+      * * * * * root /usr/local/bin/firezone-monitor/tunnel.sh 2>&1 >> /var/log/firezone_monitor/tunnel.log
+      */2 * * * * root /usr/local/bin/firezone-monitor/ping.sh 10.0.32.100 2>&1 >> /var/log/firezone_monitor/ping.log
```

```suggestion
      */2 * * * * root /usr/local/bin/firezone-monitor/ping.sh 10.0.32.100 2>&1 >> /var/log/firezone_monitor/ping_internal.log
      */10 * * * * root /usr/local/bin/firezone-monitor/ping.sh 8.8.4.4 2>&1 >> /var/log/firezone_monitor/ping_google_dns_ipv4.log
      */10 * * * * root /usr/local/bin/firezone-monitor/ping.sh 2001:4860:4860::8844 2>&1 >> /var/log/firezone_monitor/ping_google_dns_ipv6.log   
```

I just added the Google DNS as resources for this client. Probably a good idea to test that pings work for external Resources and IPv6 Resources too.

Just curious -- did the client DNS flakiness end up preventing us from having a test for that too here?

---

Ah, but you may need a `ping6.sh` script for the IPv6 one.

---

> I just added the Google DNS as resources for this client. Probably a good idea to test that pings work for external Resources and IPv6 Resources too. Ah, but you may need a ping6.sh script for the IPv6 one.

Sounds good. I can add a ping6.sh no problem

> Just curious -- did the client DNS flakiness end up preventing us from having a test for that too here?
Yeah, I was never able to reliably get DNS to work when bringing things up from scratch.  Some times the DNS would be set properly and other times it wouldn't, so I'm guessing it's some race condition with another service running on the box, but not exactly sure.

---

I just realized something about the external tests you added...we have no way to verify that the ping is going through the firezone tunnel, since these are public addresses.  If the tunnel is down all together the test will fail, because I've got a `tunnel_check` function that runs before all the tests, but if the tunnel is up, there's no way for us to tell if the packets are actually going through the tunnel right?

---

Ok, updated the tests to use the `-I` flag as we discussed in standup.  I've also added the IPv6 ping test.